tests/components/dyson/test_fan.py

```diff
@@ -320,14 +396,334 @@ def test_service_set_night_mode(self):
         self.hass.data[DYSON_DEVICES] = []
         dyson_device.entity_id = 'fan.living_room'
         self.hass.data[dyson.DYSON_FAN_DEVICES] = [dyson_device]
-        dyson.setup_platform(self.hass, None, mock.MagicMock())
+        dyson.setup_platform(self.hass, None,
+                             mock.MagicMock(), mock.MagicMock())
 
-        self.hass.services.call(dyson.DOMAIN, dyson.SERVICE_SET_NIGHT_MODE,
+        self.hass.services.call(dyson.DYSON_DOMAIN,
+                                dyson.SERVICE_SET_NIGHT_MODE,
                                 {"entity_id": "fan.bed_room",
                                  "night_mode": True}, True)
-        assert not dyson_device.night_mode.called
+        assert not dyson_device.set_night_mode.called
 
-        self.hass.services.call(dyson.DOMAIN, dyson.SERVICE_SET_NIGHT_MODE,
+        self.hass.services.call(dyson.DYSON_DOMAIN,
+                                dyson.SERVICE_SET_NIGHT_MODE,
                                 {"entity_id": "fan.living_room",
                                  "night_mode": True}, True)
-        dyson_device.night_mode.assert_called_with(True)
+        dyson_device.set_night_mode.assert_called_with(True)
+
+
+@asynctest.patch('libpurecool.dyson.DysonAccount.login', return_value=True)
+@asynctest.patch('libpurecool.dyson.DysonAccount.devices',
+                 return_value=[_get_dyson_purecool_device()])
+async def test_purecool_turn_on(devices, login, hass):
+    """Test turn on."""
+    device = devices.return_value[0]
+    dyson_parent.setup(hass, _get_config())
```

Comment:
Use `async_setup_component` from setup.py to set up the dyson component.

Reply:
Done.